MaxText/train.py

```diff
@@ -371,6 +421,23 @@ def train_loop(config, state=None):
 
     write_metrics(writer, local_metrics_file, running_gcs_metrics, metrics, step, config)
 
+    if config.eval_interval > 0 and step > start_step and step % config.eval_interval == 0:
+      assert eval_data_iterator
+      cumulative_eval_metrics = {"total_loss": 0., "total_weights": 0.}
+      for eval_batch in eval_data_iterator:
+        with mesh, nn_partitioning.axis_rules(config.logical_axis_rules):
+          _, eval_metrics = p_eval_step(
+            state, eval_batch, nextrng
+          )
+        cumulative_eval_metrics['total_loss'] += float(eval_metrics['scalar']['evaluation/total_loss'])
+        cumulative_eval_metrics['total_weights'] += float(eval_metrics['scalar']['evaluation/total_weights'])
+      eval_loss = cumulative_eval_metrics['total_loss'] / (cumulative_eval_metrics['total_weights'] + EPS)
+      max_logging.log(f"average loss after {step=}: {eval_loss=}, total_weights={cumulative_eval_metrics['total_weights']}")
+      if config.enable_mllog:
+        mllog_utils.early_stop_check(config, step, eval_loss)
+      if eval_loss <= config.target_eval_loss:
```

Comment:
Shouldn't we log here?

Reply:
I mean maxtext print!

Reply:
Good suggestion, will add maxtext print to notify early stop.